src/Fcf/Data/List.hs

```diff
@@ -242,6 +279,36 @@ type family Drop_ (n :: Nat) (xs :: [a]) :: [a] where
   Drop_ _ '[]       = '[]
   Drop_ n (x ': xs) = Drop_ (n TL.- 1) xs
 
+-- | Type-level list takeWhile.
+--
+-- === __Example__
+--
+-- >>> :kind! Eval (TakeWhile ((>=) 3) '[1, 2, 3, 4, 5])
+-- Eval (TakeWhile ((>=) 3) '[1, 2, 3, 4, 5]) :: [Nat]
+-- = '[1, 2, 3]
+data TakeWhile :: (a -> Exp Bool) -> [a] -> Exp [a]
+type instance Eval (TakeWhile p '[]) = '[]
+type instance Eval (TakeWhile p (x ': xs)) = Eval
+    (If (Eval (p x))
+        (Pure (x ': Eval (TakeWhile p xs)))
```

Comment:
There shouldn't be an `Eval` in the last two arguments of `If`.

Reply:
ah, sorry, I should have learned by now, changed, and also removed 2 spaces andmade similar changes to DropWhile